etree.go

```diff
@@ -113,6 +119,13 @@ type CharData struct {
 	whitespace bool
 }
 
+// PlainTextData represents character data within XML which will be encoded with CDATA.
+type PlainTextData struct {
+	Data       string
+	parent     *Element
+	whitespace bool
+}
+
```

Comment:
I think I would prefer to keep one type (`CharData`) for all text-related tokens.  Let's instead add a hidden `plain bool` field to the `CharData` struct.  Then modify the corresponding `CreateCharData` and `CreateTextData` methods to set the `plain` field appropriately.